js/util/script_detection.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const verticalRegExp = new RegExp([
+    '[ᄀ-ᇿ]',
+    '[가-힣]',
+    '[ㄱ-ㆎ]',
+    '[一-鿌]',
+    '[㐀-䶵]',
+    // eslint-disable-next-line no-irregular-whitespace
+    '[　-〿]',
+    '\uD840[\uDC00-\uFFFF]|[\uD841-\uD872]|\uD873[\u0000-\uDEAF]', // '[𠀀-𬺯]'
+    '[！-￮]',
+    '[ぁ-ゟ]',
+    '[゠-ヿ]',
+    '[ㇰ-ㇿ]',
+    '[ꀀ-꓆]',
+    '[᠀-ᢪ]'
```

Comment:
We should also remove Mongolian from this regular expression. Unlike the other scripts, Mongolian must be written vertically (never horizontally) and its glyphs are always rotated when set vertically. So apart from the fact that we don’t support complex text shaping yet (mapbox/mapbox-gl#4), Mongolian is already laid out correctly along lines from 3:00 to 6:00 and from 9:00 to 12:00. With this PR as is, Mongolian would only be laid out correctly from 3:00 to 4:30 and from 9:00 to 10:30.

If we want to address Mongolian, I think we should do that as tail work. But for that matter, I’m having a difficult time finding any area in OpenStreetMap or the Mapbox Streets source where roads are labeled in Mongolian. So best to leave Mongolian alone.